src/linter_test.cc

```diff
@@ -16,40 +16,100 @@
 
 #include "src/linter.h"
 
-#include "gtest/gtest.h"
+#include <string>
+#include <fstream>
+#include <sstream>
+#include <cstdio>
 
+#include "gtest/gtest.h"
 
-namespace zetasql {
+namespace zetasql::linter {
 
-namespace linter {
+namespace {
 
 TEST(LinterTest, StatementLineLengthCheck) {
     const absl::string_view sql =
         "SELECT e, sum(f) FROM emp where b = a or c < d group by x";
 
-    EXPECT_TRUE(checkLineLength(sql).ok());
-    EXPECT_FALSE(checkLineLength(sql, 10).ok());
-    EXPECT_TRUE(checkLineLength(sql, 10, ' ').ok());
+    EXPECT_TRUE(CheckLineLength(sql).ok());
+    EXPECT_FALSE(CheckLineLength(sql, 10).ok());
+    EXPECT_TRUE(CheckLineLength(sql, 10, ' ').ok());
 }
 
-
 TEST(LinterTest, StatementValidityCheck) {
     const absl::string_view valid_sql = "SELECT 5+2";
 
     const absl::string_view invalid_sql = "SELECT 5+2 sss ddd";
 
-    EXPECT_TRUE(checkStatement(valid_sql).ok());
-    EXPECT_FALSE(checkStatement(invalid_sql).ok());
+    EXPECT_TRUE(CheckStatement(valid_sql).ok());
+    EXPECT_FALSE(CheckStatement(invalid_sql).ok());
 
-    EXPECT_TRUE(checkStatement(
+    EXPECT_TRUE(CheckStatement(
         "SELECT * FROM emp where b = a or c < d group by x").ok());
 
-    EXPECT_TRUE(checkStatement(
+    EXPECT_TRUE(CheckStatement(
         "SELECT e, sum(f) FROM emp where b = a or c < d group by x").ok());
 
-    EXPECT_FALSE(checkStatement(
+    EXPECT_FALSE(CheckStatement(
         "SELET A FROM B\nSELECT C FROM D").ok());
 }
 
-}  // namespace linter
-}  // namespace zetasql
+TEST(LinterTest, SemicolonCheck) {
+    const absl::string_view valid_sql =
+        "SELECT 3+5;\nSELECT 4+6;";
+    const absl::string_view invalid_sql =
+        "SELECT 3+5\nSELECT 4+6;";
+    const absl::string_view valid_sql2 =
+        "SELECT 3+5;  \nSELECT 4+6";
+    const absl::string_view invalid_sql2 =
+        "SELECT 3+5  ;  \nSELECT 4+6";
+
+    EXPECT_TRUE(CheckSemicolon(valid_sql).ok());
+    EXPECT_FALSE(CheckSemicolon(invalid_sql).ok());
+    EXPECT_TRUE(CheckSemicolon(valid_sql2).ok());
```

Comment:
This, actually, should be not ok: the second statement doesn't end with a semicolon.

Reply:
In design doc, I read "last statement in file can be without semicolon" and allow missing semicolon for last statement. Should it remain like this or I can also check for the last statement.

Reply:
Oh, design doc was ambiguous there, my bad. What I meant, that ZetaSQL parser and analyzer do not care if there is no semicolon before the last statement, therefore we should enforce semicolon with a lint check :) Feel free to update the doc, by the way.

Reply:
Fixed, It is failing when last statement doesn't have a semicolon